ref(github): Use signed urls for external installs

AniketDas-Tekky opened the pull request:
Github -> Sentry integration installs now use signed urls.
Some other QoL improvements
- External github installs now redirect back to github once install is complete
- Added a constant for signed URL expiration

More work is needed as URL signing is done every time, instead of once.

Fixes API-1953

Commit:
ref(github): Use signed URL for github installation [WIP]

Update the github installation to use a signed URL. Unit tests are still needed

Fixes API-1956

Commit:
ref(github): Use signed URL for github installation [WIP]

Update the github installation to use a signed URL. Unit tests are still needed

Fixes API-1956

Commit:
committing changes due to incorrect branch

Commit:
committing changes due to incorrect branch

Commit:
ref(github): Use signed urls for exteral installs

Work in progress. Running into issues during the redirects

Fixes API-1953

Commit:
ref(github): Use signed urls for exteral installs

Work in progress. Running into issues during the redirects

Fixes API-1953

scefali commented on src/sentry/integrations/github/urls.py:
```diff
@@ -10,4 +12,9 @@
         name="sentry-extensions-github-search",
     ),
+    url(
+        r"^configure/(?P<signed_params>[^\/]+)/$",
```
@AniketDas-Tekky shouldn't this just be a query param: `?signed_params=< signed_params>`

Not in the URL itself, see what we do for Vercel for example.

Commit:
Fixed the URL to properly have the signed parameters in the query

Commit:
Fixed the URL to properly have the signed parameters in the query

scefali commented on src/sentry/web/frontend/pipeline_advancer.py:
```diff
@@ -40,9 +41,12 @@ def handle(self, request, provider_id):
+            )
             return self.redirect(
-                reverse("integration-installation", args=[provider_id, installation_id])
+                f"{reverse('github-integration-installation')}/?signed_params={signed_params}"
```
@AniketDas-Tekky might want to make a comment how Github is the only integration to use this which is why it's Github specific

Commit:
Properly unsign the URL during github extension install to finish the pipeline

Added a new provider for github -> sentry installs so that installs will return
to github once installs are complete

Commit:
Properly unsign the URL during github extension install to finish the pipeline

Added a new provider for github -> sentry installs so that installs will return
to github once installs are complete

Commit:
Merge branch 'signed-install-API-1953' of github.com:getsentry/sentry into signed-install-API-1953

scefali commented on src/sentry/constants.py:
```diff
@@ -552,3 +552,6 @@ def from_str(cls, string):
 DataCategory = sentry_relay.DataCategory
+
+# 24 hours to finish integration installation
+INSTALL_EXPIRATION_TIME = 60 * 60 * 24
```
nice!

scefali replied:
Mind updating other integrations to use this such as Microsoft Teams?

Commit:
Added unit tests for signed github install urls
